Mathlib/GroupTheory/Perm/Basic.lean

```diff
@@ -154,6 +154,16 @@ theorem self_trans_inv (e : Perm α) : e.trans e⁻¹ = 1 :=
 theorem symm_mul (e : Perm α) : e.symm * e = 1 :=
   Equiv.self_trans_symm e
 
+theorem trans_eq_one_iff_eq_symm (e₁ e₂ : Equiv.Perm α) : e₁.trans e₂ = 1 ↔ e₁ = e₂.symm := by
```

Comment:
Two requests:

* I think this is really just a variant of `mul_eq_one_iff_eq_inv` (on general groups) with the multiplication the other way around. Can you add that lemma?
* Can you generalize this to `Equiv` rather than `Equiv.perm`?